docs/pool_connection.md

```diff
@@ -0,0 +1,246 @@
+```
+
+## Pool
+
+### List of methods
+* [`indy_cheqd_pool_add`](#indy_cheqd_pool_add)
+* [`indy_cheqd_pool_get_config`](#indy_cheqd_pool_get_config)
+* [`indy_cheqd_pool_get_all_config`](#indy_cheqd_pool_get_all_config)
+* [`indy_cheqd_pool_broadcast_tx_commit`](#indy_cheqd_pool_broadcast_tx_commit)
+* [`indy_cheqd_pool_abci_query`](#indy_cheqd_pool_abci_query)
+* [`indy_cheqd_pool_abci_info`](#indy_cheqd_pool_abci_info)
+
+#### **indy_cheqd_pool_add**
+This method is needed for adding information about pool which will be used to connect to.
+
+Input parameters:
+* `alias` - is a human-readable string,
+* `rpc_address` - address for connecting to the node, like `http://1.2.3.4:26657`, port `26657` is default value.
+* `chain_id` - identifier of the network.
+As result structure like PoolConfig is expected in response:
+```
+{
+    "alias":"test_pool",
+    "rpc_address":"rpc_address",
+    "chain_id":"chain_id"
+}
+```
+#### **indy_cheqd_pool_get_config**
+This method is needed for getting config information about connecting to the pool by alias.
+
+Input parameters:
+* `alias` - human-readable string, represents pool alias, like `test_pool`.
+
+Expected result is structure PoolConfig, like:
+```
+{
+    "alias":"test_pool",
+    "rpc_address":"rpc_address",
+    "chain_id":"chain_id"
+}
+```
+#### **indy_cheqd_pool_get_all_config**
+The same as [indy_cheqd_pool_get_config](#indy_cheqd_pool_get_config) but returns the list of structures.
+Response should be like:
+```
+[Object({
+	"alias": String("test_pool_1"),
+	"chain_id": String("chain_id"),
+	"rpc_address": String("rpc_address")
+}), 
+Object({
+	"alias": String("test_pool_2"),
+	"chain_id": String("chain_id"),
+	"rpc_address": String("rpc_address")
+})]
+```
+#### **indy_cheqd_pool_broadcast_tx_commit**
+This method allows to send a txn to all the nodes.
+
```

Comment:
Do we really need this? It'll be impossible for a node to know where ALL nodes are, at best it will be able to know the peers connected to it. Cosmos handles this by default, so why do we need a separate call?

Reply:
Also with the way that Cosmos works, if the transaction is submitted to multiple validator nodes:

1. Will it not just appear like a duplicate transaction was attempted?
2. Is it not going to cost a LOT of gas/fees to pay to multiple nodes>